frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,188 @@
+    nav.sidenav.switcher.shouldHaveText('Administrator');
+    cy.visit('/k8s/all-namespaces/import');
+    cy.window().then((win: any) => {
+      const isLocalDevEnvironment = win.SERVER_FLAGS?.authDisabled;
+      cy.readFile(`${PLUGIN_PATH}/oc-manifest.yaml`).then((textManifest) => {
+        const yamlManifest = safeLoadAll(textManifest);
+        const yamlManifestUpdatedPullSpec = yamlManifest.map((yamlDefinition) => {
+          if (yamlDefinition.kind === 'Deployment' && !isLocalDevEnvironment && PLUGIN_PULL_SPEC) {
+            return _.defaultsDeep(
+              {},
+              {
+                spec: {
+                  template: {
+                    spec: {
+                      containers: [{ name: 'console-demo-plugin', image: PLUGIN_PULL_SPEC }],
+                    },
+                  },
+                },
+              },
+              yamlDefinition,
+            );
+          }
+          return yamlDefinition;
+        });
+        const finalYAMLText = yamlManifestUpdatedPullSpec.map((obj) => safeDump(obj)).join('---\n');
+        yamlEditor.isImportLoaded();
+        yamlEditor.setEditorContent(finalYAMLText).then(() => {
+          yamlEditor.clickSaveCreateButton();
+          cy.get(errorMessage).should('not.exist');
+          cy.byTestID('success-icon').should('have.length', 5);
+          cy.byTestID('import-more-yaml').should('exist');
+          enableDemoPlugin(true);
+        });
+      });
+    });
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    enableDemoPlugin(false);
+    cy.deleteProject(PLUGIN_NAME);
+    cy.logout();
+  });
+
+  it(`test Dynamic Nav items`, () => {
+    const dynamicNavIDs = ['1', '2'];
+    dynamicNavIDs.forEach((id) => dynamicNavTest(id));
+  });
+
+  it(`test Test Consumer nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'Test Consumer']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'Extensions of type Console.flag/Model');
+    cy.get('.pf-c-card').should('have.length', 20);
+  });
+
+  it(`test Test Utilities nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'Test Utilities']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'Utilities from Dynamic Plugin SDK');
+    cy.get('.pf-c-card__title').should('contain', 'Utility: consoleFetchJSON');
+  });
+
+  it(`test List Page nav item`, () => {
+    const podName = 'openshift-state-metrics';
+    nav.sidenav.clickNavLink(['Demo Plugin', 'List Page']);
+    listPage.titleShouldHaveText('OpenShift Pods List Page');
+    listPage.rows.shouldBeLoaded();
+    listPage.filter.byName(podName);
+    listPage.rows.shouldExist(podName);
+  });
+
+  it(`test K8s API nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'K8s API']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'K8s API from Dynamic Plugin SDK');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sCreate');
```

Comment:
It's more important to test that the utilities themselves work and don't have errors.

Reply:
@spadgett tested for content and no errData